src/cc/libclient/Writer.cc

```diff
@@ -1526,10 +1527,11 @@ class Writer::Impl :
                     return;
                 }
                 if (theStatus == kErrorReadOnly && mClosingFlag &&
-                        0 < mCloseOp.chunkId && mCloseOp.chunkVersion < 0) {
+                        0 < mCloseOp.chunkId && mKeepLeaseFlag) {
```

Comment:
The shouldn't, in theory, as in this case the statement says that the close must have been issued by this exact writer, and if it hasn't issued close then the write should fail. It would still go through the possibly redundant retries, and the fail, unless something corrupted the reply status, in which case retry logic would succeed. While the case with other concurrent writers is not supported for object store blocks, and seem implausible to succeed anyway, it is best to always enforce correctness.